### PR TITLE
Add support for Drude particles

### PR DESCRIPTION
**Pull request by peastman**
Initial implementation of #1080.  I've added support for loading Drude particles from CHARMM files and writing them to OpenMM force fields.  I'm still working on testing and making some required changes to OpenMM, but this is basically complete and functional.

### CODE REVIEW

**Comment by peastman** [comment]
If the changes look ok, this should be ready to merge.  To work correctly with OpenMM, it needs the changes from https://github.com/openmm/openmm/pull/2623.  That's why a few test cases are failing.

File: parmed/openmm/parameters.py [hunk 1]
```diff
@@ -213,6 +224,7 @@ def from_parameterset(cls, params, copy=False, remediate_residues=True):
         new_params._combining_rule = params.combining_rule
         new_params.default_scee = params.default_scee
         new_params.default_scnb = params.default_scnb
+        new_params.unique_atom_types = unique_atom_types
```
Comment (swails) [inline parmed/openmm/parameters.py:227]: `unique_atom_types` is only set in this constructor, but is used like it's always present below (unless I'm missing something).  It should probably be set to `False` in `__init__` as well to make sure the attribute is always present.

**Comment by peastman** [comment]
Is it ok if I have it use the latest dev build of OpenMM?  After the next release comes out, we can switch it back.

**Comment by swails** [comment]
Latest dev is fine in general.

There were issues with one version awhile ago which is why I switched it to
a pinned version.

But things are stable enough these days that we can just use the dev build.

**Comment by peastman** [comment]
All of the Python 3 builds are passing.  OpenMM no longer supports Python 2, so that build is failing.  Are you still trying to support it, or should I just remove it from the test matrix?

**Comment by swails** [comment]
>  Are you still trying to support it

I'm still trying to support it for now. There are still some straggler platforms out there that need to use ParmEd that are still on Python 2.7 (the Phenix refinement package is the major one, I think).

This support will likely end after AmberTools 21 is released next year.  But OpenMM is an optional dependency, so I'm fine just making the Python 2.7 build a minimal build that doesn't test with all of the optional dependencies.

**Comment by peastman** [comment]
> I'm fine just making the Python 2.7 build a minimal build that doesn't test with all of the optional dependencies.

Can I leave it to you to make whatever changes you want to the travis files?

**Comment by swails** [comment]
Done.

**Comment by peastman** [comment]
Thanks!  All tests are passing now.

**Comment by swails** [comment]
Done, thanks!

**Pull request merged**